src/pages/rooms/[id]/links/index.page.tsx

```diff
@@ -0,0 +1,147 @@
+import { GetServerSideProps, NextPage } from 'next';
+import { useState } from 'react';
+
+import { LinkManageListProps, LinkManageMode } from './links.types';
+import * as S from './links.styles';
+
+import Header from '@src/shared/components/Header';
+import { Icon } from '@src/shared/components';
+import useConfirm from '@src/shared/hooks/useConfirm';
+import LinkManageListItem from '@src/features/chat/components/LinkManageListItem';
+
+const mockImageSrc = '/images/big-chat.png';
+const MOCK_LINKS = [
+  ...[...Array(3)].map((_, i) => ({
+    id: `0${i}`,
+    imageSrc: mockImageSrc,
+    title: `제목 ${i}`,
+    date: '2022년 9월 21일 수요일',
+  })),
+  ...[...Array(5)].map((_, i) => ({
+    id: `1${i}`,
+    imageSrc: mockImageSrc,
+    title: `제목 ${i} 두줄이상 두줄이상 두줄이상 두줄이상 두줄이상 두줄이상 두줄이상 두줄이상 두줄이상`,
+    date: '2022년 9월 22일 목요일',
+  })),
+  ...[...Array(1)].map((_, i) => ({
+    id: `2${i}`,
+    imageSrc: mockImageSrc,
+    title: `제목 ${i}`,
+    date: '2022년 9월 23일 금요일',
+  })),
+  ...[...Array(4)].map((_, i) => ({
+    id: `3${i}`,
+    imageSrc: mockImageSrc,
+    title: `제목 ${i}`,
+    date: '2022년 9월 24일 토요일',
+  })),
+];
+
+const LinkManageList: NextPage<LinkManageListProps> = ({ id }) => {
+  const { confirm } = useConfirm();
+
+  const [mode, setMode] = useState<LinkManageMode>('read');
+
+  const [selectedLinkIds, setSelectedLinkIds] = useState<string[]>([]);
+
+  const handleCheckButtonClick = () => {
+    setMode('edit');
+  };
+
+  const handleCancelButtonClick = () => {
+    setMode('read');
+    setSelectedLinkIds([]);
+  };
+
+  const selectLink = (id: string) => {
+    setSelectedLinkIds((previousSelectedLinkIds) => [...previousSelectedLinkIds, id]);
```

Comment:
이 함수레벨에서 id가 중복인 경우를 걸러줄 수 없을것 같아서 코드가 좀 더러워 지더라도 요런식으로 바꿔보는건 어떤가요?
```suggestion
    setSelectedLinkIds((previousSelectedLinkIds) => Array.from(new Set([...previousSelectedLinkIds, id])));
```

Reply:
이 부분은 lodash union 사용하여 리팩토링 헀습니다~ [869e6db](https://github.com/memochat/memochat-client/pull/41/commits/869e6db93e21c1bd4d646786a7a3ebcf0c62dd68)